clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -726,7 +728,8 @@ private KafkaConsumer(ConsumerConfig config,
                     metadata,
                     time,
                     retryBackoffMs,
-                    config.getInt(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG));
+                    config.getInt(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG),
+                    Math.min(5000, heartbeatIntervalMs));
```

Comment:
Now we can change this to just `heartbeatIntervalMs`? Would be nice also to have a comment to the effect that we ensure that `poll` will not block longer than the heartbeat interval to avoid starving the heartbeat thread.